### PR TITLE
fix(invite-members): enable Resend button + display correct disabled message for Remove button

### PR DESCRIPTION
**Pull request by ameliahsu**
was displaying incorrect disabled message:
Screenshot 2024-09-16 at 2 13 28 PM

fixed by checking if the member has the `pending` property instead of checking for `inviterName`

also enabled Resend Invite button for invites sent by the current user
Screenshot 2024-09-16 at 4 06 03 PM

### CODE REVIEW

File: static/app/views/settings/organizationMembers/organizationMemberRow.tsx [hunk 1]
```diff
@@ -216,7 +213,7 @@ export default class OrganizationMemberRow extends PureComponent<Props, State> {
                     : isPartnershipUser
                       ? t('You cannot make changes to this partner-provisioned user.')
                       : // only show this message if member can remove invites but invite was not sent by them
-                        isInvite && canRemoveInvites && !isInviteFromCurrentUser
+                        pending && canInvite && !isInviteFromCurrentUser
```
Comment (cathteng) [inline static/app/views/settings/organizationMembers/organizationMemberRow.tsx:216]: does `pending` refer to whether the member has a `user` property?

Comment (ameliahsu) [reply static/app/views/settings/organizationMembers/organizationMemberRow.tsx:216]: Sorry I updated the PR description, originally I was checking if `user` was null but then I realized there was a `pending` property which is true if the invite is still pending

File: static/app/views/settings/organizationMembers/organizationMemberRow.tsx [hunk 2]
```diff
@@ -216,7 +213,7 @@ export default class OrganizationMemberRow extends PureComponent<Props, State> {
...
                         ? t('Your role cannot modify this invite.')
```
Comment (cathteng) [inline static/app/views/settings/organizationMembers/organizationMemberRow.tsx:217]: would it be better to say "You cannot modify this invite" bc `canInvite` is `True`?

File: static/app/views/settings/organizationMembers/organizationMemberRow.tsx [hunk 3]
```diff
@@ -162,7 +159,7 @@ export default class OrganizationMemberRow extends PureComponent<Props, State> {
               {isInviteSuccessful && <span>{t('Sent!')}</span>}
               {!isInviting && !isInviteSuccessful && (
                 <Button
-                  disabled={!canAddMembers}
+                  disabled={!canAddMembers && !(canInvite && isInviteFromCurrentUser)}
```
Comment (cathteng) [inline static/app/views/settings/organizationMembers/organizationMemberRow.tsx:162]: this is a bit hard to grok, would writing it as `!canAddMembers || !canInvite || !isInviteFromCurrentUser` still work?

**Review by cathteng — APPROVED** [review]
nitpicky comments

File: static/app/views/settings/organizationMembers/organizationMemberRow.tsx (hunk 3 above)
Comment (ameliahsu) [reply static/app/views/settings/organizationMembers/organizationMemberRow.tsx:162]: for me it's easier to think about when the button should be enabled, which is when either
- `canAddMembers` is true
- `canInvite` is true AND `isInviteFromCurrentUser` is true

but you're right this logic is hard to follow so I'll make a new boolean or something for readability 🫡
